modules/ml-training/k8s-managed/modulestack.yaml

```diff
@@ -0,0 +1,106 @@
+AWSTemplateFormatVersion: 2010-09-09
+Description: This stack deploys a Module specific IAM permissions
+
+Parameters:
+  DeploymentName:
+    Type: String
+    Description: The name of the deployment
+  ModuleName:
+    Type: String
+    Description: The name of the Module
+  RoleName:
+    Type: String
+    Description: The name of the IAM Role
+  DagBucketName:
+    Type: String
+    Description: The name Bucket where MWAA DAG artifacts are deployed
+  DagPath:
+    Type: String
+    Description: The path in the Dag Bucket where MWAA DAG artifacts are deployed
+  EksClusterAdminRoleArn:
+    Type: String
+    Description: Role from the EKS module with cluster admin permissions
+
+
+Resources:
+  Policy:
+    Type: 'AWS::IAM::Policy'
+    Properties:
+      PolicyDocument:
+        Statement:
+          - Action:
+              - 's3:Create*'
+              - 's3:Put*'
+              - 's3:Delete*'
+              - 's3:Get*'
+              - 's3:List*'
+            Effect: Allow
+            Resource:
+              - 'arn:aws:s3:::cdk*'
+              - !Sub "arn:aws:s3:::${DagBucketName}/${DagPath}/*"
+          - Action:
+              - 'iam:CreatePolicy'
+              - 'iam:DetachRolePolicy'
+              - 'iam:DeletePolicy'
+              - 'iam:DeleteRolePolicy'
+              - 'iam:CreateRole'
+              - 'iam:DeleteRole'
+              - 'iam:AttachRolePolicy'
+              - 'iam:PutRolePolicy'
+              - 'iam:UpdateRole'
+              - 'iam:CreatePolicyVersion'
+              - 'iam:DeletePolicyVersion'
+              - 'iam:TagRole'
+              - 'iam:UntagRole'
+              - 'iam:PassRole'
+            Effect: Allow
+            Resource:
+              - !Sub "arn:aws:iam::${AWS::AccountId}:role/addf-${DeploymentName}-${ModuleName}*"
+              - !Sub "arn:aws:iam::${AWS::AccountId}:role/service-role/addf-${DeploymentName}-${ModuleName}*"
+          - Effect: Allow
+            Action:
+              - "iam:Get*"
+              - "iam:List*"
+              - "cloudformation:Describe*"
+              - "s3:List*"
+              - 'kms:CreateKey'
+              - 'kms:List*'
+              - 'kms:Describe*'
+              - 'kms:TagResource'
+              - "cloudformation:GetTemplate"
+              - "ecr:*"
+              - "ec2:Get*"
+              - "ec2:Describe*"
+              - "elasticloadbalancing:Describe*"
+              - "logs:CreateLogGroup"
+              - "logs:Describe*"
+              - "logs:DescribeLogGroups"
+            Resource: '*'
```

Comment:
this gives the role full access to all ecr .... can you restrict further?

Reply:
will do